records/generator.go

```diff
@@ -57,8 +67,8 @@ func (rg *RecordGenerator) ParseState(c Config, masters ...string) error {
 
 // Tries each master and looks for the leader
 // if no leader responds it errors
-func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
-	var sj state.State
+func (rg *RecordGenerator) findMaster(masters ...string) (*state.State, error) {
```

Comment:
It's primarily to be able to provide a nil (Optional). I'd rather mesos-dns fail in an obvious way if somehow error checking fails, as opposed to returning an empty state, and the entire zone emptying out.